instruments/ondax/lm.py

```diff
@@ -0,0 +1,382 @@
+        control driver
+        """
+        def __init__(self, parent):
+            self.parent = parent
+
+        @property
+        def target(self):
+            """
+            The ACC target current
+            :return: Current ACC of the Laser
+            :rtype: quantities.mA
+            """
+            response = float(self.parent.query("rstli?"))
+            return response*pq.mA
+
+        def enable(self):
+            """
+            Enable the ACC driver
+            """
+            self.parent.sendcmd("lcen")
+
+        def disable(self):
+            """
+            Disable the ACC driver
+            """
+            self.parent.sendcmd("lcdis")
+
+        def on(self):
+            """
+            Turn on the ACC driver
+            """
+            self.parent.sendcmd("lcon")
+
+        def off(self):
+            """
+            Turn off the ACC driver
+            """
+            self.parent.sendcmd("lcoff")
+
+    class AutomaticPowerControl(object):
+        """
+        Options and functions related to the laser diode's automatic power
+        control driver.
+        """
+        def __init__(self, parent):
+            self.parent = parent
+
+        @property
+        def target(self):
+            """
+            Return the target laser power in mW
+            :return: the target laser power
+            :rtype: quantities.mW
+            """
+            response = self.parent.query("rslp?")
+            return float(response)*pq.mW
+
+        def enable(self):
+            """
+            Enable the APC driver
+            """
+            self.parent.sendcmd("len")
+
+        def disable(self):
+            """
+            Disable the APC driver
+            :return:
+            """
+            self.parent.sendcmd("ldis")
+
+        def start(self):
+            """
+            Start the APC scan
+            :return:
+            """
+            self.parent.sendcmd("sps")
+
+        def stop(self):
+            """
+            Stop the APC scan
+            :return:
+            """
+            self.parent.sendcmd("cps")
+
+    class Modulation(object):
+        """
+        Options and functions related to the laser's optical output modulation.
+        """
+        def __init__(self, parent):
+            self.parent = parent
+
+        @property
+        def on_time(self):
+            """
+            Get the on time of TTL modulation, in mS
+            :return: the on time in the TTL modulation.
+            :rtype: quantities.mS
+            """
+            response = self.parent.query("stsont?")
+            return float(response)*pq.mS
+
+        @on_time.setter
+        def on_time(self, newval):
+            """
+            Set the on time og TTL modulation, in mS.
+            :param newval: the modulation on time, in mS.
+            :type newval: quantities.mS
+            """
+            newval = newval.rescale('mS').magnitude
```

Comment:
`newval` needs to be wrapped in a call to the utility function `assume_units` for consistency

`newval = assume_units(newval, pq.ms).rescale('mS').magnitude`